fe/fe-core/src/main/java/org/apache/doris/analysis/TupleDescriptor.java

```diff
@@ -159,6 +164,84 @@ public void setTable(TableIf tbl) {
         table = tbl;
     }
 
+    public Set<Long> getSampleTabletIds() {
+        return sampleTabletIds;
+    }
+
+    /**
+     * First, determine how many rows to sample from each partition according to the number of partitions.
+     * Then determine the number of Tablets to be selected for each partition according to the average number
+     * of rows of Tablet,
+     * If seek is not specified, the specified number of Tablets are pseudo-randomly selected from each partition.
+     * If seek is specified, it will be selected sequentially from the seek tablet of the partition.
+     * And add the manually specified Tablet id to the selected Tablet.
+     * simpleTabletNums = simpleRows / partitionNums / (partitionRows / partitionTabletNums)
+     */
+    public void computeSampleTabletIds(List<Long> tabletIds, TableSample tableSample) {
```

Comment:
why impl this function in TupleDescriptor? since TupleDescriptor is use for describe a tuple, this is very weired.

Reply:
done